tests/common/fixtures/advanced_reboot.py

```diff
@@ -0,0 +1,510 @@
+import ipaddress
+import itertools
+import json
+import logging
+import pytest
+import time
+
+from common.errors import RunAnsibleModuleFail
+from common.mellanox_data import is_mellanox_device as isMellanoxDevice
+from common.reboot import reboot as rebootDut
+from ptf_runner import ptf_runner
+
+logger = logging.getLogger(__name__)
+
+# Globals
+HOST_MAC_COUNT = 126
+TIME_BETWEEN_SUCCESSIVE_TEST_OPER = 420
+PTFRUNNER_QLEN = 1000
+
+class AdvancedReboot:
+    '''
+    AdvancedReboot class collects information about the current testbed. This information is used by test cases to build
+    inboot/preboot list. The class transfers number of configuration files to the dut/ptf in preparation for reboot test.
+    Test cases can trigger test start utilizing runRebootTestcase API.
+    '''
+    def __init__(self, request, testbed_devices, testbed, **kwargs):
+        '''
+        Class contructor.
+        @param request: pytest request object
+        @param testbed_devices: fixture provides information about testbed devices
+        @param testbed: fixture provides information about testbed
+        @param kwargs: extra parameters including reboot type
+        '''
+        assert 'rebootType' in kwargs and kwargs['rebootType'] in ['fast-reboot', 'warm-reboot'], (
+            "Please set rebootType var."
+        )
+
+        global HOST_MAC_COUNT
```

Comment:
You shouldn't need `global` if you're only reading the variable.

Reply:
is this dependent on python version? I recall I hit an issue reading a global variable before with another PR.

Reply:
It shouldn't be, this has been the behavior going back [at least as far as version 2.0](https://docs.python.org/2.0/ref/global.html). My guess would be that the interpreter was flagging some operation as RW even though it was being used for RO?